generators/uscore-r4/generator.rb

```diff
@@ -322,69 +344,99 @@ def search_param_constants(search_parameters, sequence)
 def create_search_validation(sequence)
   search_validators = ''
   sequence[:search_param_descriptions].each do |element, definition|
+    search_validators += %(
+        when '#{element}')
     type = definition[:type]
-    contains_multiple = definition[:contains_multiple]
     path_parts = definition[:path].split('.')
-    path_parts[0] = 'resource'
     path_parts = path_parts.map { |part| part == 'class' ? 'local_class' : part }
+    path_parts.shift
     case type
-    when 'CodeableConcept'
+    when 'Period'
       search_validators += %(
-        when '#{element}'
-          codings = #{path_parts.join('&.')}#{'&.first' if contains_multiple}&.coding
-          assert !codings.nil?, '#{element} on resource did not match #{element} requested'
-          assert codings.any? { |coding| !coding.try(:code).nil? && coding.try(:code) == value }, '#{element} on resource did not match #{element} requested'
+          comparator = value[0..1]
+          value = value[2..-1] if ['ge', 'gt', 'le', 'lt', 'ne', 'sa', 'eb', 'ap'].include? comparator
+          value_found = can_resolve_path(resource, '#{path_parts.join('.')}') do |period|
+            start_date = DateTime.xmlschema(period&.start)
+            end_date = DateTime.xmlschema(period&.end)
+            value_date = DateTime.xmlschema(value)
+            case comparator
+            when 'ge'
+              end_date >= value_date || end_date.nil?
+            when 'le'
+              start_date <= value_date || start_date.nil?
+            when 'gt'
+              end_date > value_date || end_date.nil?
+            when 'lt'
+              start_date < value_date || start_date.nil?
+            when 'ne'
+              value_date < start_date || value_date > end_date
+            when 'sa'
+              start_date > value_date
+            when 'eb'
+              end_date < value_date
+            when 'ap'
+              true # don't have a good way to check this
```

Comment:
I ended up implementing it as search range overlaps with target range since that was the wording in the FHIR search page (though it also says something about 10%). Anyways, this should never be hit  anyways since we're not searching by 'ap' during testing.